cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -90,15 +91,17 @@ func NewKubeStaticPodPathManager(realDir, tempDir, backupDir, backupEtcdDir stri
 
 // NewKubeStaticPodPathManagerUsingTempDirs creates a new instance of KubeStaticPodPathManager with temporary directories backing it
 func NewKubeStaticPodPathManagerUsingTempDirs(realManifestDir string, saveManifestsDir, saveEtcdDir bool) (StaticPodPathManager, error) {
-	upgradedManifestsDir, err := constants.CreateTempDirForKubeadm("kubeadm-upgraded-manifests")
+	kubernetesTmpDir := filepath.Join(constants.KubernetesDir, "tmp")
```

Comment:
This should not be here, the idea is that `CreateTempDirForKubeadm` should attach the "tmp" portion if needed.